router/broker.go

```diff
@@ -305,47 +362,80 @@ func (b *broker) syncPublish(pub *wamp.Session, msg *wamp.Publish, pubID wamp.ID
 }
 
 func newSubscription(id wamp.ID, subscriber *wamp.Session, topic wamp.URI, match string) *subscription {
+	subscribers := map[*wamp.Session]struct{}{}
+	if subscriber != nil {
+		subscribers = map[*wamp.Session]struct{}{subscriber: {}}
+	}
+
 	return &subscription{
 		id:          id,
 		topic:       topic,
 		match:       match,
 		created:     wamp.NowISO8601(),
-		subscribers: map[*wamp.Session]struct{}{subscriber: {}},
+		subscribers: subscribers,
 	}
 }
 
-func (b *broker) syncSubscribe(subscriber *wamp.Session, msg *wamp.Subscribe, match string) {
+func (b *broker) syncSaveEvent(eventStore *eventHistoryStoreItem, pub *wamp.Publish, event *wamp.Event) {
+	item := &eventHistoryItem{
+		publication: pub,
+		event: &storedEvent{
+			timestamp:    time.Now(),
+			Subscription: event.Subscription,
+			Publication:  event.Publication,
+			Details:      event.Details,
+			Arguments:    event.Arguments,
+			ArgumentsKw:  event.ArgumentsKw,
+		},
+	}
+	eventStore.events = append(eventStore.events, item)
+
+	if eventStore.isLimitReached {
+		eventStore.events = eventStore.events[1:]
+	} else if len(eventStore.events) > eventStore.limit {
+		eventStore.isLimitReached = true
+		eventStore.events = eventStore.events[1:]
+	}
+}
+
+func (b *broker) syncInitSubscription(topic wamp.URI, match string, subscriber *wamp.Session) (*subscription, bool) {
```

Comment:
Nitpick, name the return value so that the boolean is not flapping around in the breeze like that.